main.rb

```diff
@@ -0,0 +1,101 @@
+require './app'
+
+@app = App.new
+
+OPTIONS = {
+  1 => :list_all_books,
+  2 => :list_all_people,
+  3 => :create_person,
+  4 => :create_book,
+  5 => :create_rental,
+  6 => :list_rentals_for_person,
+  7 => :exit_program
+}.freeze
+
+def main
+  display_menu
+
+  loop do
+    choice = prompt_choice
+    handle_choice(choice)
+    break if choice == 7
+  end
+end
+
+def display_menu
+  puts 'Please select an option:'
+  puts '1. List all books'
+  puts '2. List all people'
+  puts '3. Create a person'
+  puts '4. Create a book'
+  puts '5. Create a rental'
+  puts '6. List all rentals for a person'
+  puts '7. Quit'
+end
+
+def prompt_choice
+  print 'Enter your choice (1-7): '
+  gets.chomp.to_i
+end
+
+def handle_choice(choice)
+  if OPTIONS.key?(choice)
+    send(OPTIONS[choice])
+  else
+    invalid_choice
+  end
+end
+
+def list_all_books
+  @app.list_all_books
+end
+
+def list_all_people
+  @app.list_all_people
+end
+
+def create_person
+  print 'Enter the person name: '
+  name = gets.chomp
+  print 'Enter the person age: '
+  age = gets.chomp.to_i
+  print 'Enter the person type (student/teacher): '
+  type = gets.chomp.downcase
+  print 'Enter if the person has permission (y/n): '
+  parent_permission = gets.chomp.downcase == 'y'
+  @app.create_person(name, age, type, parent_permission)
+end
+
+def create_book
+  print 'Enter the book title: '
+  title = gets.chomp
+  print 'Enter the book author: '
+  author = gets.chomp
+  @app.create_book(title, author)
+end
+
+def create_rental
+  print 'Enter the book title: '
```

Comment:
- [ ] If you look to the video you can see if user wants to create a new rental. The have options to pick rather than writing the name of book or person. Imagine you have 10 or 20 of books to rent but you need to go back to list of books to see the name of book and create a new rental. This is not a good practice. Please update rental and make sure it's identical with the original design. Thank you!